samples/src/main/java/com/backblaze/b2/sample/B2Sample.java

```diff
@@ -247,6 +247,13 @@ private static void mainGuts(PrintWriter writer,
         B2FileVersion file2again = client.getFileInfo(file2.getFileId());
         writer.println("file2Again: " + file2again);
 
+        // get file info by name.
+        B2FileVersion file2ByName = client.getFileInfoByName(bucket.getBucketName(), file2.getFileName());
+        writer.println("file2ByName: " + file2ByName);
+        if (!file2again.equals(file2ByName)) {
+            throw new RuntimeException("file!");
+        }
+
```

Comment:
thanks.  have you run B2Sample?  is this check passing?

Reply:
Nope. But I fixed it.

Reply:
hi, valentin.

my goal with this check is to ensure that the answers *do* match.  it looks like you've changed from 'equals()' to checking individual parts of the answer.  in particular, it looks like you've skipped checking fileInfo.  it looks to me like the issue is that B2Headers.getB2FileInfo() is looking for fileInfo headers case-sensitively.  i'll work on that and provide a fix.  after that, you should be able to put this back to an "equals()" check.

thanks,
ab

Reply:
equals not possible. Because server with HEAD request doesn't return fileinfo. 

B2FileVersion{fileId='4_zb5bf041b3fe70c616a370e14_f116d66bad903d3c8_d20180427_m083556_c002_v0001094_t0020', contentLength=261989, contentType='application/pdf', contentSha1='d6507d49b1822065c8d82423610d35659c760678', action='null', uploadTimestamp=1524818156000, fileInfo=[0], fileName='casio_4778.pdf'}

Reply:
And I'm not sure, it will be possible pass equals with no modification on server side response for HEAD request.